### ISSUE TITLE
awaitCameraForCoordinates with MapCameraManagerDelegate receiver

### ISSUE THREAD

**Issue by alibagherifam** [mapbox/mapbox-maps-android#2414]
## Current situation
The extension function `awaitCameraForCoordinates` has a `MapboxMap` receiver. However, it simply calls `cameraForCoordinates`, which is actually a function defined within the `MapCameraManagerDelegate` interface.

## The problem
It is possible to call `cameraForCoordinates` on a `MapCameraManagerDelegate` interface (which is more abstract than the `MapboxMap` object). But calling the suspend version is only possible on a `MapboxMap` object.

```kotlin
val map: MapboxMap = /* init */
val cameraManager: MapCameraManagerDelegate = map

// Ok
map.cameraForCoordinates(/* ... */)

// Ok
cameraManager.cameraForCoordinates(/* ... */)

// Ok
map.awaitCameraForCoordinates(/* ... */)

// Error
cameraManager.awaitCameraForCoordinates(/* ... */)
```

## The ideal solution
It would be nice if `awaitCameraForCoordinates` were an extension function with a `MapCameraManagerDelegate` receiver.

```kotlin
// From
suspend fun MapboxMap.awaitCameraForCoordinates(/* .. */)

// To 
suspend fun MapCameraManagerDelegate.awaitCameraForCoordinates(/* .. */)
```

**Comment by kiryldz** [comment]
@alibagherifam thanks for the suggestion! We'll take a look.

**Comment by natiginfo** [comment]
@alibagherifam we have addressed this suggestion in 11.6.0-beta.1 release. Please, try it out. Closing this ticket for now.

**Comment by alibagherifam** [comment]
@natiginfo Thank you for the quick fix! The new receiver for `awaitCameraForCoordinates` enhances the flexibility and usability of the API.
